sql/core/src/main/scala/org/apache/spark/sql/execution/joins/JoinCodegenSupport.scala

```diff
@@ -42,13 +42,15 @@ trait JoinCodegenSupport extends CodegenSupport with BaseJoinExec {
       buildRow: Option[String] = None): (String, String, Seq[ExprCode]) = {
     val buildSideRow = buildRow.getOrElse(ctx.freshName("buildRow"))
     val buildVars = genOneSideJoinVars(ctx, buildSideRow, buildPlan, setDefaultValue = false)
+    val streamVars2 = streamVars.map(_.copy())
     val checkCondition = if (condition.isDefined) {
       val expr = condition.get
       // evaluate the variables from build side that used by condition
-      val eval = evaluateRequiredVariables(buildPlan.output, buildVars, expr.references)
+      val eval = evaluateRequiredVariables(buildPlan.output ++ streamPlan.output,
```

Comment:
You know, for consistency with code below I should really make this `streamPlan.output ++ buildPlan.output` and `streamVars2 ++ buildVars`.

Reply:
Also I should update the comment above this line.